### PR TITLE
[chore][receiver/pulsar]: add observability to pulsar receiver

### PR DESCRIPTION
**Pull request by VihasMakwana**
- Enhancement

This PR adds observability in pulsar receiver.

### CODE REVIEW

**Comment by VihasMakwana** [comment]
@dmitryax can you take a look here?

**Review by dao-jun — APPROVED** [review]

**Comment by VihasMakwana** [comment]
@dmitryax can you take a look?

File: receiver/pulsarreceiver/pulsar_receiver.go [hunk 1]
```diff
@@ -302,11 +340,11 @@ func consumeLogsLoop(ctx context.Context, c *pulsarLogsConsumer) error {
 			_ = c.consumer.Ack(message)
 			return err
```
Comment (atoulme) [inline receiver/pulsarreceiver/pulsar_receiver.go:341]: you need to end the logsOp here too, and with any early return.

File: receiver/pulsarreceiver/pulsar_receiver.go [hunk 2]
```diff
@@ -202,10 +228,11 @@ func consumeMetricsLoop(ctx context.Context, c *pulsarMetricsConsumer) error {
...
```
Comment (atoulme) [inline receiver/pulsarreceiver/pulsar_receiver.go:229]: need to end the trace there too

Comment (VihasMakwana) [reply receiver/pulsarreceiver/pulsar_receiver.go:235]: Done

File: receiver/pulsarreceiver/pulsar_receiver.go (hunk 1 above)
Comment (VihasMakwana) [reply receiver/pulsarreceiver/pulsar_receiver.go:350]: Done.

**Review by Isma1982 — APPROVED** [review]

File: receiver/pulsarreceiver/pulsar_receiver.go [hunk 3]
```diff
@@ -81,13 +94,16 @@ func consumerTracesLoop(ctx context.Context, c *pulsarTracesConsumer) error {
...
+		obsCtx := c.obsrecv.StartTracesOp(ctx)
 		message, err := c.consumer.Receive(ctx)
 		if err != nil {
 			if strings.Contains(err.Error(), alreadyClosedError) {
+				c.obsrecv.EndTracesOp(obsCtx, unmarshaler.Encoding(), 0, err)
 				return err
 			}
 			if errors.Is(err, context.Canceled) {
 				c.settings.Logger.Info("exiting consume traces loop")
+				c.obsrecv.EndTracesOp(obsCtx, unmarshaler.Encoding(), 0, err)
```
Comment (dmitryax) [inline receiver/pulsarreceiver/pulsar_receiver.go:106]: Shouldn't this be moved under `if err != nil {` so we close the trace on any failure?

Comment (VihasMakwana) [reply receiver/pulsarreceiver/pulsar_receiver.go:106]: Oh yeah. That's actually better.

Comment (VihasMakwana) [reply receiver/pulsarreceiver/pulsar_receiver.go]: Done. PTAL!

**Comment by VihasMakwana** [comment]
@dmitryax can we merge this one?

**Review by atoulme — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**